agent/xds/resources.go

```diff
@@ -0,0 +1,62 @@
+package xds
+
+import (
+	"fmt"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/hashicorp/go-hclog"
+
+	"github.com/hashicorp/consul/agent/proxycfg"
+)
+
+// ResourceGenerator is associated with a single gRPC stream and creates xDS
+// resources for a single client.
+type ResourceGenerator struct {
+	Logger         hclog.Logger
+	CheckFetcher   HTTPCheckFetcher
+	CfgFetcher     ConfigFetcher
```

Comment:
aside: In a future PR I want to move `HTTPCheckFetcher` and `ConfigFetcher` over into `agent/proxycfg` so that the only blob of data we need in `agent/xds` is the `ConfigSnapshot`.